rama-http-backend/src/client/proxy/layer/proxy_connector/connector.rs

```diff
@@ -64,6 +77,45 @@ impl InnerHttpProxyConnector {
         self,
         stream: S,
     ) -> Result<upgrade::Upgraded, HttpProxyError> {
+        let response = match self.version {
+            Some(Version::HTTP_10 | Version::HTTP_11) => {
+                Self::handshake_h1(self.req, stream).await?
+            }
+            Some(Version::HTTP_2) => Self::handshake_h2(self.req, stream).await?,
+            None => match self.req.version() {
```

Comment:
You really do want `http1` to be the default.

You can still achieve this though with your current code.
E.g. you can default the version instead of `None` to `Some(Version::HTTP_10)`.

And than add these setter methods:

```
pub(super) fn with_version(mut self, version: Version) -> Self {
    self.version = Some(version);
    self
}

pub(super) fn set_version(&mut self, version: Version) -> &mut Self {
    self.version = Some(version);
    self
}

pub(super) fn with_auto_version(mut self) -> Self {
    self.version = None;
    self
}

pub(super) fn set_auto_version(&mut self) -> &mut Self {
    self.version = None;
    self
}
```